python/ray/data/dataset.py

```diff
@@ -338,6 +335,10 @@ def parse_filename(row: Dict[str, Any]) -> Dict[str, Any]:
             num_gpus: The number of GPUs to reserve for each parallel map worker. For
                 example, specify `num_gpus=1` to request 1 GPU for each parallel map
                 worker.
+            concurrency: The number of Ray workers to use concurrently. For a fixed-sized
+                workers pool of size ``n``, specify ``concurrency=n``. For a autoscaling
+                workers pool from ``m`` to ``n`` workers, specify ``concurrency=(m, n)``.
+                ``None`` to use the default value of system.
```

Comment:
```suggestion
            concurrency: The number of Ray workers to use concurrently. For a fixed-sized
                worker pool of size ``n``, specify ``concurrency=n``. For an autoscaling
                worker pool from ``m`` to ``n`` workers, specify ``concurrency=(m, n)``.
                ``None`` to use the default value of system.
```

It would be good to specify how the default value is chosen.

Reply:
We can add a section about concurrency in the doc, and add a link here.

Reply:
Updated the wording per @stephanie-wang, and updated the doc per @raulchen.